### ISSUE TITLE
Trying to view bee nests with Jade protocol enabled kicks the player with IndexOutOfBoundsException in the console log

### ISSUE THREAD

**Issue by QuanTrieuPCYT** [LeavesMC/Leaves#73]
### 服务端版本 LeavesMC Version

git-Leaves-"4de2295"

### 预期的行为 Expected behavior

Viewing a bee nest with `jade-protocol`  enabled in `leaves.yml` should report the bee count inside that nest to players with the Jade mod installed on their client-side.

### 观测到的实际行为 Observed/Actual behavior

Instead, the server kicks the player out with "Invalid custom payload!", and the console outputs this error:

```
[11:42:29] [Server thread/ERROR]: Couldn't dispatch custom payload
java.lang.IndexOutOfBoundsException: readerIndex(9) + length(1) exceeds writerIndex(9): PooledUnsafeDirectByteBuf(ridx: 9, widx: 9, cap: 9)
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1442) ~[netty-buffer-4.1.87.Final.jar:4.1.87.Final]
	at io.netty.buffer.AbstractByteBuf.readByte(AbstractByteBuf.java:730) ~[netty-buffer-4.1.87.Final.jar:4.1.87.Final]
	at net.minecraft.network.FriendlyByteBuf.readByte(FriendlyByteBuf.java:1284) ~[?:?]
	at net.minecraft.network.FriendlyByteBuf.readVarInt(FriendlyByteBuf.java:589) ~[?:?]
	at net.minecraft.network.FriendlyByteBuf.readEnum(FriendlyByteBuf.java:575) ~[?:?]
	at net.minecraft.network.FriendlyByteBuf.readBlockHitResult(FriendlyByteBuf.java:835) ~[?:?]
	at top.leavesmc.leaves.protocol.JadeProtocol.requestTileData(JadeProtocol.java:359) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at top.leavesmc.leaves.protocol.JadeProtocol.handlePacket(JadeProtocol.java:125) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.handleCustomPayload(ServerGamePacketListenerImpl.java:3582) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundCustomPayloadPacket.handle(ServerboundCustomPayloadPacket.java:37) ~[?:?]
	at net.minecraft.network.protocol.game.ServerboundCustomPayloadPacket.handle(ServerboundCustomPayloadPacket.java:15) ~[?:?]
	at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:51) ~[?:?]
	at net.minecraft.server.TickTask.run(TickTask.java:18) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
	at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
	at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1344) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:198) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
	at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1321) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1314) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
	at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1292) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1180) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[leaves-1.20.1.jar:git-Leaves-"4de2295"]
	at java.lang.Thread.run(Thread.java:1589) ~[?:?]
[11:42:29] [Server thread/INFO]: QuanTrieuPCYT lost connection: Invalid custom payload!
[11:42:29] [Server thread/INFO]: QuanTrieuPCYT left the game
```

### 复现步骤 Steps/models to reproduce

Install Jade on the client-side, enable `jade-protocol` in `leaves.yml`, install the Jade mod on the client side and join the server.

Pick a random bee nest and try to hover your crosshair into it

### 插件与数据包列表。 Plugin and Datapack List.

`Server Plugins (0):`

**Comment by s-yh-china** [comment]
this issue is happened after the updated leaves version?

**Comment by QuanTrieuPCYT** [comment]
> this issue is happened after the updated leaves version?

Not really, I just tried Leaves for the first time and that happened

**Comment by WhitePhosphor** [comment]
same problem
Maybe the nbt tag is too long？
---- Replied Message ----
| From | ***@***.***> |
| Date | 07/28/2023 13:01 |
| To | LeavesMC/Leaves ***@***.***> |
| Cc | Subscribed ***@***.***> |
| Subject | Re: [LeavesMC/Leaves] Trying to view bee nests with Jade protocol enabled kicks the player with IndexOutOfBoundsException in the console log (Issue #73) |

this issue is happened after the updated leaves version?

**Comment by s-yh-china** [comment]
Can you provide your Jade version and the block you look to?

**Comment by QuanTrieuPCYT** [comment]
Currently on 11.3.0

**Comment by s-yh-china** [comment]
i cant reproduce it .?
if possible, can I access your server for testing

**Comment by QuanTrieuPCYT** [comment]
> i cant reproduce it .? if possible, can I access your server for testing

Not really for now, it's a local machine and I can't really port forward.

**Comment by QuanTrieuPCYT** [comment]
Yeah I think I found the issue, should be unrelated to this fork. It's ViaVersion + ViaBackwards failing to report NBT data to the client.

I joined the server with a 1.19.4 client with Via installed, tried wandering around and noticed one thing: Jade didn't seem to report entities when you hover to them, only blocks. Tested it with a 1.19.4 server and the behavior produced as expected, I think imma file a report to them then.

Joined a Leaves-powered server with 1.20 Jade and things seem to be working real fine
